packages/codegen-ui-react/lib/react-component-render-helper.ts

```diff
@@ -324,19 +325,46 @@ export function getSyntaxKindToken(operator: RelationalOperator): BinaryOperator
   }
 }
 
-export function getConditionalOperandExpression(operand: string | number | boolean): Expression {
-  switch (typeof operand) {
+export function getConditionalOperandExpression(
+  operand: string | number | boolean,
+  operandType: string | undefined,
+): Expression {
+  if (typeof operand === 'string' && operandType && operandType !== 'string') {
+    return valueToCorrectLiteral(operand, operandType);
+  }
+  return valueToJsxLiteral(operand);
+}
+
+function valueToCorrectLiteral(value: any, valueType: string): PrimaryExpression {
```

Comment:
perhaps rename to `typedValueToLiteral`?

Reply:
fixed naming to distinguish between the two functions better